### DISCUSSION TITLE
Automatic Same Name

### DISCUSSION THREAD

**Discussion by Jay123210599** [ImageMagick/ImageMagick#7783] [Help]
### ImageMagick version

7.1.1-40

### Operating system, version and so on

Windows 11

### Description

When doing any operation in ImageMagick, how do I automatically make the output file have the same name as the input file?

For example, I convert a png file to jpg, and I want the output jpg to have the same name as the input png.

Using a script is not allowed. I mean I want to put in something that will get me the same name for the output from the input, like %d or something.

**Comment by dceeee** [comment]
magick input.png -set filename: %t %[filename:].jpg

**Reply by Jay123210599** [reply]
What's %t % and [filename:] for? Do I have to add -set filename?

**Reply by fmw42** [reply]
see https://imagemagick.org/script/escape.php

```
-set filename:xxx sets xxx to the name of the file (use any name for xxx, such as just f as filename:f)
%t sets the name to the name of the file excluding suffix. 
%f sets the name to exactly the input (including suffix)
%[filename:xxx] tells IM to use xxx as part of your output name
```

**Reply by Jay123210599** [reply]
So I only use %t?

**Reply by fmw42** [reply]
Depends what suffix you want?  If you want something other than what is the input, then add .suffix. If you want the same, then use %f with no .suffix added.

**Reply by Jay123210599** [reply]
Do I need to use `-set filename` with %f? I want the output to have a different suffix than the input (e.g. png to jpg).

**Reply by fmw42** [reply]
I explained that above.  In that case, use %t and add .suffix.

`magick input.suffix -set filename:f %t %[filename:f].new_suffix`

**Reply by Jay123210599** [reply]
Is there a way do this _without_ the `-set filename`?

**Reply by fmw42** [reply]
No. Not that I know about.

**Comment by snibgo** [comment]
@Jay123210599: dceeee has suggested a command for you. If you put it in a BAT script, you need to double the percentage signs. Does the command do what you want?
